libs/libc/stdio/lib_fclose.c

```diff
@@ -115,32 +115,18 @@ int fclose(FAR FILE *stream)
 
       nxmutex_unlock(&slist->sl_lock);
 
-      /* Check that the underlying file descriptor corresponds to an an open
-       * file.
-       */
-
-      if (stream->fs_fd >= 0)
-        {
-          /* Close the file descriptor and save the return status */
+      /* Close the file descriptor and save the return status */
 
-#ifdef CONFIG_FDSAN
-          uint64_t tag;
-          tag = android_fdsan_create_owner_tag(ANDROID_FDSAN_OWNER_TYPE_FILE,
-                                               (uintptr_t)stream);
-          status = android_fdsan_close_with_tag(stream->fs_fd, tag);
-#else
-          status = close(stream->fs_fd);
-#endif
+      status = stream->fs_iofunc.close(stream->fs_cookie);
```

Comment:
should we check close != NULL before calling, or fill the empty callback function in fopencookie

Reply:
Added NULL check. Fopencookie actually allows the user to have close == NULL.